.jenkins/deploy_agent.Jenkinsfile

```diff
@@ -0,0 +1,119 @@
+@Library("OpenEnclaveCommon") _
+oe = new jenkins.common.Openenclave()
+
+// The below timeout is set in minutes
+GLOBAL_TIMEOUT = 240
+
+def ACCDeployVM() {
+    stage("Deploy ${AGENT_NAME}") {
+        node("nonSGX") {
+            timeout(GLOBAL_TIMEOUT) {
+                cleanWs()
+                checkout scm
+                dir("${WORKSPACE}/.jenkins/provision") {
+                    oe.azureEnvironment("./deploy-agent.sh", OE_DEPLOY_IMAGE)
+                }
+            }
+        }
+    }
+}
+
+def generateInventory() {
+    if (agent_type == "windows") {
+        sh """
+           echo "[windows-agents]" > inventory/hosts
+           echo ${AGENT_NAME.toLowerCase()}.${REGION}.cloudapp.azure.com >> inventory/hosts
+           """
+    } else {
+        sh """
+           echo "[linux-agents]" > inventory/hosts
+           echo ${AGENT_NAME.toLowerCase()}.${REGION}.cloudapp.azure.com >> inventory/hosts
+           """
+    }
+}
+
+def generateVariablesFile() {
+    def var_file = "inventory/host_vars/${AGENT_NAME.toLowerCase()}.${REGION}.cloudapp.azure.com"
+    if (agent_type == "windows") {
+      sh """{
+         echo jenkins_agent_name: ${AGENT_NAME}
+         echo jenkins_agent_label: ${AGENT_LABEL}
+         echo jenkins_url: ${JENKINS_PRIVATE_URL}
+         echo jenkins_admin_name: ${JENKINS_ADMIN_NAME}
+         echo jenkins_admin_password: ${JENKINS_ADMIN_PASSWORD}
```

Comment:
Do we have to worry about any of these secret variables showing up in plaintext in the logs when this job gets run?

Reply:
Jenkins is automatically masking secrets that come from withCredentials block, they are masked with *** in the logs.